File: frontend/models/ClimateReading.js
import mongoose from 'mongoose'

const ClimateReadingSchema = new mongoose.Schema({
  timestamp: { type: Date, required: true },
  source: { type: String, required: true },
  country: {
    name: { type: String, required: true },
    isoCode: { type: String, required: true },
    region: { type: String, required: true },
    coordinates: {
      lat: { type: Number, required: true },
      lng: { type: Number, required: true }
    }
  },
  metrics: {
    temperature: { type: Number, required: true },
    co2Level: { type: Number, required: true },
    humidity: Number,
    seaLevel: Number,
    precipitation: Number,
    windSpeed: Number,
    uvIndex: Number,
    airQualityIndex: Number
  },
  anomaly: {
    detected: { type: Boolean, default: false },
    severity: String,
    description: String
  },
  processed: { type: Boolean, default: true }
}, {
  timestamps: true,
  collection: 'climatereadings'
})

export default mongoose.models.ClimateReading || mongoose.model('ClimateReading', ClimateReadingSchema)

File: backend/models/climate_reading.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime


class Coordinates(BaseModel):
    lat: float
    lng: float


class Country(BaseModel):
    name: str
    isoCode: str
    region: str
    coordinates: Coordinates


class Metrics(BaseModel):
    temperature: float
    co2Level: float
    humidity: Optional[float] = None
    seaLevel: Optional[float] = None
    precipitation: Optional[float] = None
    windSpeed: Optional[float] = None
    uvIndex: Optional[float] = None
    airQualityIndex: Optional[float] = None


class Anomaly(BaseModel):
    detected: bool = False
    severity: Optional[str] = None
    description: str = ""


class ClimateReadingCreate(BaseModel):
    timestamp: datetime
    source: str
    country: Country
    metrics: Metrics
    anomaly: Anomaly = Anomaly()
    processed: bool = True


class ClimateReadingOut(BaseModel):
    id: str = Field(alias="_id")
    timestamp: datetime
    source: str
    country: Country
    metrics: Metrics
    anomaly: Anomaly
    processed: bool = True
    createdAt: Optional[datetime] = None
    updatedAt: Optional[datetime] = None

    class Config:
        populate_by_name = True
